modules/catalog/src/test/java/org/apache/ignite/internal/catalog/CatalogZoneTest.java

```diff
@@ -0,0 +1,464 @@
+        assertEquals("expression", zone.filter());
+        assertEquals("test_profile", zone.storageProfiles().profiles().get(0).storageProfile());
+    }
+
+    @Test
+    public void testSetDefaultZone() {
+        CatalogZoneDescriptor initialDefaultZone = latestActiveCatalog().defaultZone();
+
+        // Create new zone
+        {
+            StorageProfileParams storageProfile = StorageProfileParams.builder()
+                    .storageProfile("test_profile")
+                    .build();
+
+            CatalogCommand createZoneCmd = CreateZoneCommand.builder()
+                    .zoneName(TEST_ZONE_NAME)
+                    .storageProfilesParams(List.of(storageProfile))
+                    .build();
+
+            assertThat(manager.execute(createZoneCmd), willCompleteSuccessfully());
+
+            assertNotEquals(TEST_ZONE_NAME, latestActiveCatalog().defaultZone().name());
+        }
+
+        // Set new zone as default.
+        {
+            CatalogCommand setDefaultCmd = AlterZoneSetDefaultCommand.builder()
+                    .zoneName(TEST_ZONE_NAME)
+                    .build();
+
+            int prevVer = latestActiveCatalog().version();
+
+            assertThat(manager.execute(setDefaultCmd), willCompleteSuccessfully());
+            assertEquals(TEST_ZONE_NAME, latestActiveCatalog().defaultZone().name());
+
+            // Make sure history has not been affected.
+            Catalog prevCatalog = Objects.requireNonNull(manager.catalog(prevVer));
+            assertNotEquals(TEST_ZONE_NAME, prevCatalog.defaultZone().name());
+            assertNotEquals(latestActiveCatalog().defaultZone().id(), prevCatalog.defaultZone().id());
+        }
+
+        // Create table in the new zone.
+        {
+            assertThat(manager.execute(simpleTable(TABLE_NAME)), willCompleteSuccessfully());
+
+            Catalog catalog = latestActiveCatalog();
+            CatalogTableDescriptor tab = Objects.requireNonNull(manager.table(TABLE_NAME, catalog.time()));
+
+            assertEquals(catalog.defaultZone().id(), tab.zoneId());
+        }
+
+        // Setting default zone that is already the default changes nothing.
+        {
+            int lastVer =  manager.latestCatalogVersion();
+
+            CatalogCommand setDefaultCmd = AlterZoneSetDefaultCommand.builder()
+                    .zoneName(TEST_ZONE_NAME)
+                    .build();
+
+            assertThat(manager.execute(setDefaultCmd), willCompleteSuccessfully());
+            assertEquals(lastVer, manager.latestCatalogVersion());
+        }
+
+        // Drop old default zone.
+        {
+            CatalogCommand dropCommand = DropZoneCommand.builder()
+                    .zoneName(initialDefaultZone.name())
+                    .build();
+
+            assertThat(manager.execute(dropCommand), willCompleteSuccessfully());
+        }
+    }
+
+    @Test
+    public void testDropZone() {
```

Comment:
Let's also check that non-empty zone can't be dropped.

Reply:
Such test case already exists in `DropZoneCommandValidationTest::rejectToDropZoneWithTable`